pkg/mutation/path/path.go

```diff
@@ -0,0 +1,32 @@
+package path
+
+// Parse parses the string representing a path specification
+// and returns the path splitted as path entries
+func Parse(string) ([]Entry, error) {
+	return nil, nil
+}
+
+//Type is the type of a path
+type Type string
+
+// Entry represent a segment of a path
+type Entry interface {
+	Type() Type
+}
+
+// Object is a segment of a path which is not a list
+type Object struct {
+	// TypeName is the type of Entry
+	Type Type
+	// The name of the field corresponding to the next segment of the path.
+	// Must be an empty string in case
+	PointsTo *string
+}
+
+// List is a segment of a path that relates to a list
+type List struct {
+	Type     Type
+	KeyField string
+	KeyValue *string
```

Comment:
I'm still not entirely sure we need missing vs. empty string differentiation. My understanding is that a keyed list can never use an empty value for its key field.

Reply:
Yes, makes sense to me.
@maxsmythe can you provide us more details, or should this become a string? In case of globbed we have the explicit flag anyway.

Reply:
@shomron Is that an expectation or an explicit requirement in the conventions?

Unless the syntax is specifically disallowed, we should support it in order to avoid leaving any users who happen to hit this edge case out in the cold.

Reply:
I don't see it explicitly disallowed in the api-conventions. In practice they seem to be required fields, but I can't prove that's applied consistently and am happy to play it safe :)
```
 k apply -f empty.yaml
The Pod "emptystuff" is invalid:
* spec.volumes[0].name: Required value
* spec.containers[0].name: Required value
```